File: LAB4/Ex_4.1.3.py
"""
04.1.3 Shapes. Write a program that takes as an input an integer number n and prints a square and a 
rhombus filled with asterisks (*), with each side long n asterisks. Example: using n=4, the program 
shows: 
****
****
****
****
   *
  ***
 *****
*******
 *****
  ***
   *

"""
"""
Square pattern
"""
rows = int(input("Please enter the number of rows: "))
columns = int(input("Please enter the number of columns: "))

for i in range(rows):
    for j in range(columns):
        print("*", end = "")
    print()
    
    
"""
Star pattern
"""

def main(row):
    column = int(row)
    i = 0
    while (i < column):
        j = column
        while (j > i+1):
            print(" ", end="")
            j -= 1
        k = 0
        while (k <= i):
            print("*", end="")
            k += 1
        l = 0
        while (l < i):
            print("*", end="")
            l += 1

        print()
        i += 1

    i1 = 0
    while (i1 < column - 1):
        j1 = 0
        while (j1 <= i1):
            print(" ", end="")
            j1 += 1

        k1 = column - 1
        while (k1 > i1):
            print("*", end="")
            k1 -= 1

        l1 = column - 2
        while (l1 > i1):
            print("*", end="")
            l1 -= 1
        print()
        i1 += 1


if __name__ == "__main__":
    row = int(input("Please enter the number of rows: "))
    main(row)

File: LAB4/Ex_4.1.3.java
/*
Write a program that takes as an input an integer number n and prints a square and a 
rhombus filled with asterisks (*), with each side long n asterisks. Example: using n=4, the program shows:
   
   *
  ***
 *****
*******
 *****
  ***
   *

*/
import java.util.Scanner;

public class Ex_4.1.3{
  public static void main(String[] args){
    Scanner scan = new Scanner(System.in);
    
    
    int row = scan.nextInt();
    int column = row;
    
    for(int i = 0; i<row; i++){
      for(int j = column; j >=i; j--){
      System.out.print(" ");   
      }
      for(int z = 0; z<=i; z++){
        System.out.print("*");
        }
      for(int k = 0; k<i; k++){
        System.out.print("*");
       }
      System.out.println();
    }//end of first outer loop
    
    for(int i = 0; i<row-1; i++){
      for(int j = i; j<column-1; j++){
      System.out.print(" ");
      }
      for(int k = 0; k<row-2; k++){
      System.out.print("*");
      }
      for(int j = 0; j<column-2; j++){
        System.out.print("*");
       }
    System.out.println();  
    }//end of second outer for loop
  
  
  
  }//end of main
}//end of class
